lib/Fmt_odoc.ml

```diff
@@ -120,50 +118,62 @@ let list_block_elem elems f =
       in
       f elem $ break)
 
-(* Format each element with [fmt_elem] *)
-let fmt_styled style fmt_elem elems =
-  let s =
-    match style with
-    | `Bold -> "b"
-    | `Italic -> "i"
-    | `Emphasis -> "e"
-    | `Superscript -> "^"
-    | `Subscript -> "_"
-  in
-  wrap "{" "}"
-    (str_normalized s $ fmt_if_not_empty elems "@ " $ list elems "" fmt_elem)
-
-let rec fmt_inline_element : inline_element -> Fmt.t = function
-  | `Space _ -> fmt "@ "
-  | `Word w ->
-      (* Escape lines starting with '+' or '-' *)
-      let escape =
-        fmt_if_k
-          (String.length w > 0 && Char.(w.[0] = '+' || w.[0] = '-'))
-          (if_newline "\\")
-      in
-      escape $ str_normalized w
-  | `Code_span s ->
-      let s = escape_brackets s in
-      hovbox 0 (wrap "[" "]" (str_normalized ~escape:escape_brackets s))
-  | `Raw_markup (lang, s) ->
-      let lang =
-        match lang with Some l -> str_normalized l $ str ":" | None -> noop
-      in
-      wrap "{%%" "%%}" (lang $ str s)
-  | `Styled (style, elems) ->
-      fmt_styled style (ign_loc ~f:fmt_inline_element) elems
-  | `Reference (_kind, ref, txt) ->
-      let ref = fmt "{!" $ fmt_reference ref $ fmt "}" in
-      if List.is_empty txt then ref
-      else wrap "{" "}" (ref $ fmt "@ " $ fmt_inline_elements txt)
-  | `Link (url, txt) -> (
-      let url = wrap "{:" "}" (str_normalized url) in
-      match txt with
-      | [] -> url
-      | txt -> wrap "{" "}" (url $ fmt "@ " $ fmt_inline_elements txt) )
+let space_elt : inline_element with_location =
+  Location_.(at (span []) (`Space ""))
 
-and fmt_inline_elements txt = list txt "" (ign_loc ~f:fmt_inline_element)
+let rec fmt_inline_elements elements =
+  let wrap_elements opn cls ~always_wrap hd = function
+    | [] -> wrap_if always_wrap opn cls hd
+    | tl -> wrap opn cls (hd $ fmt_inline_elements (space_elt :: tl))
```

Comment:
Can you add a test case where `space_elt` is useful ? Eg. ` ... {b <margin here>+ 2} ...`

Reply:
done